Add prototype integration API. (#16538)
ckeditor/ckeditor5: docs/getting-started/setup/menubar.md

	Before adding a feature to the menu bar, make sure the plugin for that feature is added in the editor configuration.
</info-box>

## Menu bar components in custom plugins

To add menu bar components in a custom plugin, follow these steps:

> 1. Add the Menu Bar Component: Use the `editor.ui.componentFactory` to register your custom component.
> 1. Implement the Component Logic: Define the behavior and appearance of your component.
> 1. Use `editor.ui.addMenuBarItem()` method so your component won't require adding to editor config. Follow the same convention as in `config.menuBar.addItems`.

Reviewer: ```suggestion
1. Create a new UI component using `editor.ui.componentFactory`. Define its behavior and appearance.
1. Use `editor.ui.addMenuBarItem()` method to add your component to the menu bar at a desired position.
```

Additionally, API mentions (\`...\`) should link to proper API docs pages.

> Here's an example of a custom plugin that adds a button to the menu bar:

Reviewer: ```suggestion
Here's an example of a custom plugin that adds a button to the menu bar inside the "Format" menu, after the "Bold" button:
```

```js
import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
import ButtonView from '@ckeditor/ckeditor5-ui/src/button/buttonview';

class MyCustomPlugin extends Plugin {
	init() {
		const editor = this.editor;

> 		// Register the toolbar button

Reviewer: ```suggestion
		// Register the toolbar button.
```

> 		editor.ui.componentFactory.add('myCustomButton', locale => {

Reviewer 2: ```suggestion
		editor.ui.componentFactory.add( 'myCustomButton', locale => {
```
Please remember our code style.

> 		editor.ui.componentFactory.add( 'myCustomButton', locale => {

Reviewer: I think it might be nice if people follow our convention with `menuBar:` prefix?

			const view = new ButtonView(locale);

			view.set( {
				label: 'My Custom Button',
				withText: true,
				tooltip: true
			} );

> 			// Execute a command when the button is clicked

Reviewer: ```suggestion
			// Execute a command when the button is clicked.
```

			view.on( 'execute', () => {
				editor.execute('myCustomCommand');
			} );

			return view;
		} );

> 		// Register your component in preferred default position

Reviewer: ```suggestion
		// Add your component in the preferred position.
```
